lib/galaxy/tools/__init__.py

```diff
@@ -289,26 +294,40 @@ def tools_by_id(self):
 
     def get_cache_region(self, tool_cache_data_dir):
         if tool_cache_data_dir not in self.cache_regions:
-            self.cache_regions[tool_cache_data_dir] = create_cache_region(tool_cache_data_dir)
+            try:
+                with open(os.path.join(tool_cache_data_dir, 'cache.json')) as cache_file:
+                    self.cache_regions[tool_cache_data_dir] = json.load(cache_file)
```

Comment:
Can this JSON file become too large to always have it loaded in memory as part of `app.toolbox` ? I liked having a database-backed cache :pensive:

Reply:
Yeah, I'm sorry, I should have put this back into WIP. I have another variant that uses hdf5py that could work, but ... looks like that also has problems with multiple readers and writers. I am now considering writing a single JSON file per tool or debugging the diskcache approach ... but I don't have a good lead on how to do that.